src/Microsoft.AspNetCore.Mvc.TagHelpers/LinkTagHelper.cs

```diff
@@ -341,7 +358,59 @@ private void BuildFallbackBlock(TagHelperContent builder)
                 .AppendHtml("\",");
 
             AppendFallbackHrefs(builder, fallbackHrefs);
-            builder.AppendHtml("</script>");
+
+            builder.AppendHtml(", \"");
+
+            // Perf: Avoid allocating enumerator
+            for (var i = 0; i < attributes.Count; i++)
+            {
+                var attribute = attributes[i];
+                if (string.Equals(attribute.Name, HrefAttributeName, StringComparison.OrdinalIgnoreCase))
+                {
+                    continue;
+                }
+
+                attribute.WriteTo(StringWriter, HtmlEncoder);
+                StringWriter.Write(' ');
+            }
+
+            var stringBuilder = StringWriter.GetStringBuilder();
+            var scriptTags = stringBuilder.ToString();
+            stringBuilder.Clear();
+            var encodedScriptTags = JavaScriptEncoder.Encode(scriptTags);
+            builder.AppendHtml(encodedScriptTags);
+
+            builder.AppendHtml("\");</script>");
+        }
+
+        private bool HasRelStylesheet(TagHelperAttributeList attributes)
+        {
+            TagHelperAttribute relAttribute;
+            if (!attributes.TryGetAttribute(RelAttributeName, out relAttribute) ||
+                relAttribute.Value == null)
+            {
+                return false;
+            }
+
+            var attributeValue = relAttribute.Value;
+            var contentValue = attributeValue as IHtmlContent;
+            var stringValue = attributeValue as string;
+            if (contentValue != null)
+            {
+                contentValue.WriteTo(StringWriter, HtmlEncoder);
+                stringValue = StringWriter.ToString();
+
+                // Reset writer
+                StringWriter.GetStringBuilder().Clear();
+            }
+            else if (stringValue == null)
+            {
+                stringValue = attributeValue.ToString();
```

Comment:
I wasn't pushing back on doing the runtime checks, just rehashing the history. Great we're all in agreement on not doing the CSS selector restrictions. As I said in my final message yesterday, I am fine doing the "ceremony" and just wasn't sure we'd decided on that.